pymeasure/display/curves.py

```diff
@@ -85,7 +85,7 @@ def __init__(
         self.yend = getattr(self.results.procedure, self.y + "_end")
         self.ystep = getattr(self.results.procedure, self.y + "_step")
         self.ysize = int(np.ceil((self.yend - self.ystart) / self.ystep))
-        self.img_data = np.zeros((self.ysize, self.xsize))
+        self.img_data = np.full((self.ysize, self.xsize), np.nan, dtype=float)
```

Comment:
Pre-calculate bins (related to the other suggestion about the digitizing)

```suggestion

        self.xbins = np.linspace(self.xstart, self.xend, self.xsize + 1)
        self.ybins = np.linspace(self.ystart, self.yend, self.ysize + 1)
        self.img_data = np.full((self.ysize, self.xsize), np.nan, dtype=float)

```

Reply:
@CasperSchippers pushed the suggestion in [#6f39762](https://github.com/pymeasure/pymeasure/pull/1341/commits/6f39762e17804b953a3869aeabccd4e4ce115a2f)